### PR TITLE
Update User Guide - Interactive legend for TOC layers

### PR DESCRIPTION
**Pull request by ElenaGallo**
## Description

Interactive legend for TOC layers on [Display Layer settings](https://docs.mapstore.geosolutionsgroup.com/en/latest/user-guide/layer-settings/#display](https://mapstore--10477.org.readthedocs.build/en/10477/user-guide/layer-settings/#display-tab-for-other-layer-types)) section and [Catalog](https://mapstore--10477.org.readthedocs.build/en/10477/user-guide/catalog/#advanced-settings_1) section

### CODE REVIEW

File: docs/user-guide/layer-settings.md [hunk 1]
```diff
@@ -104,6 +104,16 @@ When the *Use cache options* is enabled, more controls are enabled so that it is
...
 
+* Enable/disable the **Interactive legend**. If this option is enable, filter by legend can be applied by clicking on legend items of this layer from the TOC. An example can be the following one:
+
+<video class="ms-docimage"  style="max-width:700px;" controls><source src="../img/layer-settings/interactive-legend.mp4"></video>
+
+!!! Note
+    Any type of filter rather than the legend filter will keep the legend filter beside it.
+
+!!!Warning
+    The *Interactive legend* option os available only for the layers added from WMS/CSW and WFS catalog sources.
```
Comment (allyoucanmap) [inline docs/user-guide/layer-settings.md:115]: I think we should mention that this option is experimental. I don't think it's supported by the WFS catalog yet.

```suggestion
    The *Interactive legend* is an experimental option and it is disabled by default. The flag `miscSettings.experimentalInteractiveLegend` inside [localConfig](local-config.md) must be set to true to enable this functionality. The *Interactive legend* is available only for GeoServer WMS layers that support the json format for GetLegendGraphic requests.
```

File: docs/user-guide/layer-settings.md [hunk 2]
```diff
@@ -104,6 +104,16 @@ When the *Use cache options* is enabled, more controls are enabled so that it is
 !!!warning
     The Gridset compatibility check made by MapStore whose result is shown by the Info tooltip, is usually quite reliable but should be considered anyway only to provide general matching indicators aimed at highlighting possible compatibility issues between the current layer/map settings and the remote Tile Grid. Due to the cache tolerance considered on the server side by GWC, it might even happen in some cases that the settings available on the client side don't HIT the tile cache even if all the checks listed are successful. At the same time, when the standard gridset is used, gridsets check may fail even if all WMS request are effectively HITTING the cache (e.g. because the WMTS reports a list of origins).
 
+* Enable/disable the **Interactive legend**. If this option is enable, filter by legend can be applied by clicking on legend items of this layer from the TOC. An example can be the following one:
```
Comment (allyoucanmap) [inline docs/user-guide/layer-settings.md:107]:
```suggestion
* Enable/disable the **Interactive legend**. If this option is enabled, legend entries with an associated valid filter are displayed as toggleable items inside the TOC, an user can click on these legend items to filter the content of the layer. An example can be the following one:
```

File: docs/user-guide/layer-settings.md [hunk 3]
```diff
@@ -104,6 +104,16 @@ When the *Use cache options* is enabled, more controls are enabled so that it is
...
+
+<video class="ms-docimage"  style="max-width:700px;" controls><source src="../img/layer-settings/interactive-legend.mp4"></video>
+
+!!! Note
+    Any type of filter rather than the legend filter will keep the legend filter beside it.
```
Comment (allyoucanmap) [inline docs/user-guide/layer-settings.md:112]: What do we want to highlight with this note?

**Review by allyoucanmap — CHANGES_REQUESTED** [review]

**Review by allyoucanmap — APPROVED** [review]

**Comment by tdipisa** [comment]
@ElenaGallo can you please backport this to 2024.02.xx?

**Pull request merged**